### DISCUSSION TITLE
using pyboard.py on windows

### DISCUSSION THREAD

**Discussion by pvoytas** [micropython/micropython#10084] [Using MicroPython]
Hi,
I'm using a pyboard lite v 1.0 under windows 10. Things work as expected if I edit main.py and reboot. Things work as expected if I connect using a serial term to COM4. However, I was trying to user pyboard.py and I am not being successful. The python I have on may windows machine is using anaconda so I've been working from the anaconda3 command prompt.
If I issue;
python pyboard.py -h 
I get the help for pyboard.py
If I issue
python pyboard.py -d COM4 -c 'print(1+1)'
I get no response, the command session hangs (no response to Ctl C, D, Y or Z) and I have to close the anaconda prompt window and start over.
Likewise if I try
python pyboard.py -d COM4 test.py
where test.py just tries to blink an LED, I again get the same hanging behavior.

I assume this has something to do with communicating over COM4, but the lack of error messages has me stumped.
Or, I suppose it could be using the anaconda terminal?
I would be grateful for any help getting pyboard.py working on windows 10.

Thanks!
-Paul

**Comment by kjm1102** [comment]
Have you tried a vanilla repl like thonny? Might help validate the pyboard is working OK?

**Reply by karfas** [reply]
@kjm1102: The board seems to be working, see the original post:
> Things work as expected if I edit main.py and reboot. 
> Things work as expected if I connect using a serial term to COM4.

Regarding thonny to verify the board's function: The REPL window in thonny is neither a "REPL" (the REPL runs on the board) **nor** a "vanilla" connection (thonny tries some funny things here to get to a prompt), as was pointed out in numerous other posts regarding thonny connection problems.

**Reply by pvoytas** [reply]
@kjm1102 : @karfas is right: the REPL mode works fine (I use TeraTerm for serial connection); editing main.py works fine. It is just the pyboard.py tool ( https://docs.micropython.org/en/latest/reference/pyboard.py.html ) that I can't get to work yet.

**Comment by Josverl** [comment]
Have you tried using mpremote ?
```
# add pip to conda 
conda install pip 
# install mpremote 
pip install mpremote
# list devices 
mpremote devs 
# connect to first device 
mpremote repl
# help
mpremote --help
```
